### PR TITLE
Don't modify my metadata object

### LINKED ISSUE #754: Don't modify my metadata object
### Environment Details

* SDV version: 0.14.0 (and previous)

### Description

Right now, if I pass in a [`metadata.Table` object](https://sdv.dev/SDV/api_reference/metadata/table.html) into a model and fit it, the process will modify my metadata (add extra fields to it).

This can lead to unexpected behavior due to the extra fields that were added, especially if I want to re-use the metadata object in a different model.

Can we make a deep copy of the `metadata.Table` object instead?

### Steps to reproduce

See the code below for unexpected usage

```python
from sdv.tabular import GaussianCopula
from sdv.demo import load_tabular_demo

metadata, data = load_tabular_demo('student_placements', metadata=True)

# create a model that estimates everything as a gamma distribution
model = GaussianCopula(table_metadata=metadata,
                       categorical_transformer='label_encoding',
                       default_distribution='gamma')
model.fit(data)
print(model.get_distributions()) # each distribution correctly is a Gamma

# create a new model that estimates everything as a beta
model2 = GaussianCopula(table_metadata=metadata,
                        categorical_transformer='label_encoding',
                        default_distribution='beta')

model2.fit(data)
print(model2.get_distributions()) # Everything is still gamma. This is unexpected!
```

This is happening because, unbeknownst to me, my metadata is modified to include a 'gamma' distribution field for each and every column. I am unknowingly passing it into `model2`, which is favoring the metadata over the `default_distribution` parameter.

### PR DESCRIPTION
**Pull request by amontanez24**
resolves #754

### CODE REVIEW

**Review by katxiao — APPROVED** [review]
lgtm!

File: sdv/lite/tabular.py [hunk 1]
```diff
@@ -49,15 +49,16 @@ def __init__(self, name=None, metadata=None, constraints=None):
                           'detected from your data. This process may not be accurate. '
                           'We recommend writing metadata to ensure correct data handling.')
 
-        if metadata is not None and constraints is not None:
+        else:
```
Comment (csala) [inline sdv/lite/tabular.py:52]: Wouldn't this now fail if `metadata` is `None` and `constraints` are not `None`?

File: tests/integration/tabular/test_base.py [hunk 2]
```diff
@@ -26,6 +27,37 @@ def _isinstance_side_effect(*args, **kwargs):
...
+                           categorical_transformer='label_encoding',
+                           default_distribution='gamma')
+    model.fit(data)
+    model2 = GaussianCopula(table_metadata=metadata,
+                            categorical_transformer='label_encoding',
+                            default_distribution='beta')
+    model2.fit(data)
+
+    # Assert
+    assert model._metadata != metadata
```
Comment (katxiao) [inline tests/integration/tabular/test_base.py:56]: Could we also assert that the different metadata have the expected distributions? Just to be sure.

**Review by katxiao — APPROVED** [review]
LGTM with a comment

**Review by katxiao — APPROVED** [review]

**Pull request merged**